crypto/ring-signature/src/amount/mod.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! A commitment to an output's amount, denominated in picoMOB.
```

Comment:
Yeah so... the reason this happened is:

* Currently `TokenId` is the argument to the `GeneratorCache` object, and creates some type-safety. (Perhaps, `GeneratorCache` does not need to be moved though. Otoh perhaps we want the `generators` function to also take a `TokenId` argument, for type safety?)
* Currently, the `RingSigner` trait refers to `SignableInputRing` which contains `InputSecret` which contains `Amount`.

```
/// The secrets needed to create a signature that spends an existing output as
/// an input
#[derive(Clone, Debug, Zeroize)]
#[zeroize(drop)]
pub struct InputSecret {
    /// Represents either the one-time private key, or data to derive it
    pub onetime_key_derive_data: OneTimeKeyDeriveData,
    /// The amount of the output
    pub amount: Amount,
    /// The blinding factor of the output we are trying to spend
    pub blinding: Scalar,
}
```

We want that this `Signer` stuff does not depend on transaction core, otherwise the point of this PR is defeated because you have to port all of transaction core to the hardware wallet.

I think we could go either way here, I'm fine to remove token id and amount from this crate and make `InputSecret` represent `TokenId` using a plain `u64` if people think that's better. The way I tried to do it in this PR is `Amount` is in the ring-signature crate, and `MaskedAmount` is in `transaction-core`